### ISSUE TITLE
unused-import when readline is imported to enhance input()

### ISSUE THREAD

**Issue by michalfabik** [PyCQA/pylint#8652]
### Bug description

```python
#pylint: disable=missing-docstring

import readline

input()
```

### Command used

```shell
pylint-3 a.py
```

### Pylint output

```shell
************* Module a
a.py:4:0: W0611: Unused import readline (unused-import)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

    --------------------------------------------------------------------
    Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)  
  

"If the [readline](https://docs.python.org/3/library/readline.html#module-readline) module was loaded, then [input()](https://docs.python.org/3/library/functions.html?highlight=input#input) will use it to provide elaborate line editing and history features."  (https://docs.python.org/3/library/functions.html?highlight=input#input)

### Pylint version

```shell
$ pylint-3 --version
pylint 2.14.4
astroid 2.12.14
Python 3.11.2 (main, Feb  8 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)]
$
```

### OS / Environment

Fedora 37

**Comment by DanielNoord** [comment]
I also think this is hard to code, on the other hand this is the `stdlib` so should be "fixed" in `pylint` rather than a plugin.

**Comment by Pierre-Sassoulas** [comment]
It's true that it's in the stdlib so we should do something about it (probably). Without checking the full code of ``input()`` and detecting side effects (too much work clearly, and performance would be awful) we could create an option with a list of function that have side effect and the import it impacts (?).
